finality-aleph/src/abft/common.rs

```diff
@@ -25,23 +25,39 @@ fn exponential_slowdown(
 }
 
 pub type DelaySchedule = Arc<dyn Fn(usize) -> Duration + Sync + Send + 'static>;
+pub type RecipientCountSchedule = Arc<dyn Fn(usize) -> usize + Sync + Send + 'static>;
 
 pub fn unit_creation_delay_fn(unit_creation_delay: UnitCreationDelay) -> DelaySchedule {
-    Arc::new(move |t| {
-        if t == 0 {
-            Duration::from_millis(2000)
-        } else {
-            exponential_slowdown(t, unit_creation_delay.0 as f64, 5000, 1.005)
-        }
+    Arc::new(move |t| match t {
+        0 => Duration::from_millis(2000),
+        _ => exponential_slowdown(t, unit_creation_delay.0 as f64, 5000, 1.005),
+    })
+}
+
+pub fn coord_request_delay_fn() -> DelaySchedule {
+    Arc::new(|t| match t {
+        0 => Duration::from_millis(0),
+        1 => Duration::from_millis(50),
+        2 => Duration::from_millis(1000),
+        _ => Duration::from_millis(3000 * (t as u64 - 2)),
     })
 }
 
+pub fn coord_request_recipients_fn() -> RecipientCountSchedule {
+    Arc::new(|t| if t <= 2 { 3 } else { 1 })
```

Comment:
It would be nice to have some explanation for these functions in comments, as they are written it's very unclear what they are doing. In particular this one seems weird intuitively, since we are first trying to ask more nodes, and if that fails we start asking fewer? Although I might be misunderstanding what `t` is.

Reply:
All those functions are taken from aleph-bft. The q is then do we want to make them public there or duplicate here?

Reply:
Changed to default.